### PR TITLE
Subtract merge layer

### PR DESCRIPTION
**Pull request by ericwu09**
With the merge method being deprecated by end of 8/2017, there needs to be a way to define a subtraction of two layers. This Subtract layer takes exactly two tensors and returns the difference.

### CODE REVIEW

File: keras/layers/merge.py [hunk 1]
```diff
@@ -211,6 +211,36 @@ def _merge_function(self, inputs):
         return output
 
 
+class Subtract(_Merge):
+    """Layer that subtracts two inputs.
+
+    It takes as input a list of tensors of size 2,
+    both of the same shape, and returns
```
Comment (Dref360) [inline keras/layers/merge.py]: Not necessarily the same shape?
```python
x = K.ones([10,10])
y = K.ones([1,10])
x - y # Does not raise.
```

Comment (ericwu09) [inline keras/layers/merge.py]: Thanks, added support and unit test for shape mismatch.

File: keras/layers/merge.py [hunk 2]
```diff
@@ -211,6 +211,39 @@ def _merge_function(self, inputs):
...
+    a single tensor (also of the same shape).
```
Comment (fchollet) [inline keras/layers/merge.py]: Explicitly state that it returns `inputs[0] - inputs[1]`.

Comment (ericwu09) [inline keras/layers/merge.py]: Thanks, updated the documentation.

**Comment by ericwu09** [comment]
All the tests passed, but the pull request failed. Is this related to this request, or possibly another issue?

**Comment by fchollet** [comment]
It's a temporary file system-related issue with Travis. Cause unclear; maybe Travis itself, or a recent update to pytest.

**Pull request merged**